logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -146,7 +146,12 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
 
     @Override
     public synchronized void close() throws IOException {
```

Comment:
@robbavey this is more of a general comment, but this `synchronized` seems very dangerous. If this is actually needed because we could have concurrent `close` calls, it might be safer to simply turn the `open` field into an `AtomicBoolean` and guard by atomic `open.compareAndSet(true, false)` around the whole method instead of `synchronized` to avoid any locking in the exception handling?